docs/content/guides/developer/getting-started/data-serving.mdx

```diff
@@ -0,0 +1,115 @@
+
+:::
+
+### JSON-RPC
+
+You can currently get real-time or historical data from a Sui Full node. Retention period for historical data depends on the [pruning strategy](/guides/operator/data-management.mdx#sui-full-node-pruning-policies) implemented by a node operator, though currently all Full nodes are by default configured to implicitly fall back on a [Mysten Labs managed key-value store for historical transaction data](/guides/operator/data-management.mdx#sui-full-node-key-value-store-backup).
+
+:::caution
+
+Websockets based JSON RPCs `suix_subscribeEvent` and `suix_subscribeTransaction` were deprecated in July 2024. You should not rely on those RPCs in your applications. Refer to [this section](#future-state-data-interfaces) to learn about a future alternative.
+
+:::
+
+### Custom indexer
+
+If you need object version history or another type of historical data where you may not fully rely on a Full node or RPC provider, or if you have specific query patterns that could be best served from a relational database, then you can setup your own [custom indexer](/guides/developer/advanced/custom-indexer.mdx) or reach out to a [Data indexer operator](https://github.com/sui-foundation/awesome-sui?tab=readme-ov-file#indexers--data-services) who may already have setup one. 
+
+If you setup your own indexer, you are responsible for its ongoing maintenance and the related infrastructure & operational costs. You can reduce your costs by implementing a pruning strategy for the relational database by taking into account the retention needs of your application.
+
+## Future data access interfaces
+
+Primary interfaces to access Sui access data in future will be:
+
+- Full node gRPC API will replace JSON-RPC on Full nodes, and is currently available in alpha. If you already use the JSON-RPC or are starting to utilize it as a dependency for your use case, you would need to migrate to gRPC or GraphQL (see below) within a reasonable duration. Refer to the high-level timeline for gRPC availability [below](#grpc-api).
+- [GraphQL RPC with Indexer 2.0](/references/sui-graphql.mdx) will include a performant and scalable implementation of the indexer framework and is currently available in beta. You would be able to use the indexer to load and manage data in a postgres relational database, and use a lightweight GraphQL RPC server that will be used to read data from that database.
+
+You would still be able to utilize a [RPC provider](https://sui.io/developers#dev-tools) (filter by `RPC`) or a [Data indexer operator](https://github.com/sui-foundation/awesome-sui?tab=readme-ov-file#indexers--data-services) with the above options.
+
+### gRPC API
+
+As mentioned above, gRPC API will replace the JSON-RPC on Full nodes, such that JSON-RPC will be deprecated when gRPC API is generally available. Apart from the message & request format changes between the two, the gRPC API will come with a couple of key functional differences:
```

Comment:
The functionality that we currently offer in JSON-RPC will end up split between gRPC and GraphQL (the archival/historical object access is a good example of this), this is slightly different from the fact that after all is said and down, fullnodes will only speak gRPC.

Reply:
I've already mentioned something related to that in the second point below. Beyond that I think we should figure out a proper historical data access architecture before being specific here.